code/weapon/weapon.h

```diff
@@ -303,6 +303,15 @@ enum class HomingAcquisitionType {
 	RANDOM,
 };
 
+struct ConditionalImpact {
+	particle::ParticleEffectHandle effect;
+	float min_health_threshold;
+	float max_health_threshold;
+	float min_angle_threshold;
+	float max_angle_threshold;
```

Comment:
Minor, but these could benefit from units, 0-1 factor, and degrees, etc.

Reply:
Good idea.